vllm/executor/mp_distributed_executor.py

```diff
@@ -108,6 +108,10 @@ def _init_executor(self) -> None:
         all_kwargs = []
         distributed_init_method = get_distributed_init_method(
             get_ip(), get_open_port())
+        # Get cpus_allow_list for main process and its ranks
+        import psutil
+        cpus_allow_list = psutil.Process().cpu_affinity()
```

Comment:
Why not getting ```cpus_allow_list``` in the ```CPUWorker```? I think worker children processes have same CPU affinity with the parent before the OMP thread binding.

Reply:
@bigPYJ1151  
I tested it on my EMR env. only parent process got right cpus_allow_list. children processes got 0 for cpus_allow_list and failed.  I highlight related logs in red below.
![image](https://github.com/user-attachments/assets/6c33f151-ab91-471d-a9d8-1b35ed51556e)

Therefore, I moved the codes to executor under parent process.

Reply:
I think it is due to the fork method.
Can you please help to try again with ```VLLM_WORKER_MULTIPROC_METHOD=spawn```?
If it works, you can add it to the CPU platform settings:
https://github.com/vllm-project/vllm/blob/5a8641638a5ec6cf089490d6b7f474b135ff590e/vllm/platforms/cpu.py#L147-L168

Reply:
thanks. it works. change default value from fork to spawn in envs.py

Reply:
addressed it accordingly